packages/react-client/src/ReactFlightReplyClient.js

```diff
@@ -229,6 +241,24 @@ export function processReply(
         });
         return serializeFormDataReference(refId);
       }
+      if (value instanceof Map) {
+        const partJSON = JSON.stringify(Array.from(value), resolveToJSON);
```

Comment:
~question: What's the purpose of the `Array.from` call here? I thought we already can stringify iterables so this seems redundant. Tests are passing without it.~ We `Array.from` it later anyway after we extracted the iterator so we might as well do it here since we know the value is iterable.

Reply:
It's also to avoid this being an infinite loop, since otherwise we'd just end up back here since Map is preferred over the Iterable path.

Reply:
From my local testing, dropping it here was fine, but not on Server. Or vice versa, I don't quite remember. Either way, the way you proposed makes it easier to follow.